src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -779,6 +779,8 @@ def resolve(self, logical_plan: LogicalPlan) -> SnowflakePlan:
 
         if self.subquery_plans:
             result = result.with_subqueries(self.subquery_plans)
+            if isinstance(logical_plan, SelectStatement):
```

Comment:
we probably should handle this for Selectable in general, not just SelectStatement